vis_gen.py

```diff
@@ -0,0 +1,565 @@
+            srcpath = os.path.join(train_dir_abs, filename)
+            shutil.move(srcpath, test_dir_abs)
+        print('Train test split successfully created.')
+
+        # Check number of classes in each split
+        class_train = set([name.split('_')[0] for name in
+                           os.listdir(train_dir_abs)])
+        class_test = set([name.split('_')[0] for name in
+                          os.listdir(test_dir_abs)])
+        no_missing_class = class_test.issubset(class_train)
+        print('Training dataset has {} categories.'.format(len(class_train)))
+        print('Test dataset has {} categories.'.format(len(class_test)))
+        print('All test categories in training data: {}'
+              .format(no_missing_class))
+
+        return num_train, num_test
+
+
+
+    def __get_filename(self, code_point):
+        """Get the filename for code point under current context. Filename is
+        'CODEPOINT_FONTNAME_FONTSTYLE_ANTIALIAS'
+
+        Args:
+            code_point: Single Unicode code point
+
+        Returns:
+            filename: Str, corresponding file name
+        """
+        filename = 'U+' + str(hex(ord(code_point)))[2:]
+        filename += '_'
+        filename += self.__font_name
+        filename += '_'
+        if self.__font_style:
+            filename += self.font_style
+            filename += '_'
+        filename += self.antialias
+        filename += '.png'
+        return filename
+
+
+    def __check_create_out_dir_abs(self, out_dir_abs):
+        """Check if the given absolute path exists and create if not.
+
+        Args:
+            out_dir_abs
+
+        Returns:
+            None
+
+        Raises:
+            OSError: if specified directory cannot be created
+        """
+        # Check if out_dir exists and create
+        if not os.path.isdir(out_dir_abs):
+            print("{} does not exist, creating directory..."
+                  .format(out_dir_abs))
+            try:
+                os.mkdir(out_dir_abs)
+            except OSError:
+                print("Creation of directory {} failed."
+                      .format(out_dir_abs))
+                raise
+            else:
+                print("New directory successfully created.")
+
+    def __position_text(self, context, text, x, y):
+        """Put text in the intended position (relative to the origin)."
+        
+        Args:
+            context: Qahirah context for the selected font and text size
+            text: Unicode codepoint for the text (character)
+            x: X coordinate for the intended position
+            y: Y coordinate for the intended position
+        """
+        # Calculate how much to move on x and y axis
+        text_extents = context.text_extents(text)
+        delta_x = x - text_extents.x_bearing
+        delta_y = y - text_extents.y_bearing
+
+        # Move in context
+        context.move_to((delta_x, delta_y))
+
+    def __center_text(self, context, text):
```

Comment:
write test for this by mocking context.{text_extends, move_to} see https://docs.python.org/3/library/unittest.mock.html about how to mock context